app/services/authentication/authenticator.rb

```diff
@@ -147,7 +147,7 @@ def update_profile_updated_at(user)
     end
 
     def account_less_than_a_week_old?(user, logged_in_identity)
-      provider_created_at = user.public_send(provider.user_created_at_field)
+      provider_created_at = user.identities.find_by(provider: provider.name)&.created_at
```

Comment:
Someone with more knowledge of our authenticator classes should probably sanity check this, cc @rhymes.

Reply:
This raises some questions in me. I was testing some things in BHC and noticed one of my accounts was being flagged as spam (I noticed because the Slack notification worker was failing in Sidekiq due to missing credentials).

**Question:** Is this method supposed to return true if the account is less than a week old **but** also attempt to rely on the `created_at` details that we get from the OAuth payload (meaning that if GitHub says the account is older we should use that instead of the `created_at` from the Identity model)?

Correct me if I'm wrong, but from the logic in this method I understand that the query in this line will fetch the `created_at` date of the Identity model itself, instead of the actual date provided by us from the OAuth payloads. So every newly registered user would be marked as a possible spammer because their Identity would be brand new ([saved in this line](https://github.com/forem/forem/blob/citizen428/remove-provider-created-at-columns-11414/app/services/authentication/authenticator.rb#L47)).

If the answer to the **highlighted question** above is _yes_, then I think we are not meeting the second part of the logic ("trust users with old Twitter/GitHub/etc accounts").

I may be splitting hairs here (let me know if I am and all of this isn't necessary) but here are some other follow up ideas/questions that could be worth considering in case we want to take a moment to better understand this logic:
- Apple indeed doesn't provide this `created_at` date in their payload. [We are defaulting to `Time.current`](https://github.com/forem/forem/blob/citizen428/remove-provider-created-at-columns-11414/app/services/authentication/authenticator.rb#L159-L169) and that might not be the best solution (because Apple authenticated accounts will always be marked as possible spammers)
   - To check this I ran a quick console query in BCH (easier for me than a Blazer query in DEV for JSON payloads). We would have to double check Facebook auth too so we can be sure Apple is the only way.
   - Maybe worth considering to [change the default here](https://github.com/forem/forem/blob/citizen428/remove-provider-created-at-columns-11414/app/services/authentication/authenticator.rb#L167) to `nil`? This way we can rely on another fallback and "trust Apple Auth users" rather than distrust them by default?
- To be sure how old the account is, maybe we should check for _"the oldest identity associated with this user"_?
   - This comes from thinking about users that just connected a second OAuth provider. They are now logging in with this new provider, and since the `created_at` timestamp of this Identity model in the DB is recent they are now marked as possible spammers
   - This circles back to question above about the date we are using to identify how old the account is
   - In this case a user that registered with Twitter a year ago and just now connected GitHub would technically be as old as _"the oldest identity associated with this user"_

What throws me off a bit is that the method called `social_auth_registration_recent?` ([actually discussed above as well](https://github.com/forem/forem/pull/13264/files#r607471158)) was not being used in here.

Reply:
I was under the impression we only care when they created their account with us, but you raise some good points and maybe that's not true. 🤔 

>  So every newly registered user would be marked as a possible spammer

This is, in fact, a bit how we treat them. See e.g. https://github.com/forem/forem/pull/12925 where we added a lower rate limit for comment creation for new users (account with us less than 3 days old). 

> To be sure how old the account is, maybe we should check for "the oldest identity associated with this user"?

Just to be clear, the old code also didn't do this, no behavior changed in this regard (except maybe for provider account age vs local account age), correct?

Reply:
> Just to be clear, the old code also didn't do this, no behavior changed in this regard (except maybe for provider account age vs local account age), correct?

That's correct, and given that fact I'm thinking this is more of a non-blocking question/suggestion rather than a concern.

Just FYI I'm pinging the team in charge of monitoring the `Slack::Messengers::PotentialSpammer` to give them a nudge and see if this is even a concern at all or not (this is the only place where its called). That way they can make the call of whether the potential spammer alert logic is ok or if it's worth to put it through the RFC process (for refactoring)

Reply:
Again, I'm not sure I understand all the ramifications but it seems like we're going from "tell me when the user registered themselves on GitHub" to "when did they last registered with GitHub on Forem" (especially because a person with multiple identities can connect adn disconnect them, leaving that `created_at` to actually mean "the last time they connected their OAuth identity with us". 

For example, this is when I registered on GitHub:

```ruby
> gi.auth_data_dump.dig("extra", "raw_info", "created_at")
=> "a long time ago in UTC :-D" 
```

this is the data I last attached my GitHub account to my local Forem:

```ruby
> gi.created_at.iso8601
=> "2021-02-05T14:35:26Z"
```

By changing this logic if I were to disconnect and reconnect the account, would I be branded as a spammer automatically?